torch/mps/__init__.py

```diff
@@ -0,0 +1,88 @@
+r"""
+This unit enables an interface for accessing MPS backend in python
+"""
+import torch
+from functools import lru_cache
+import threading
+from .. import Tensor
+import torch._C
+
+_initialized = False
+_initialization_lock = threading.Lock()
+default_mps_generator: torch._C.Generator = None  # type: ignore[assignment]
+
+def init():
+    r"""Initialize PyTorch's MPS state.
+    Does nothing if the MPS state is already initialized.
+    """
+    _lazy_init()
+
+def _lazy_init():
+    global _initialized
+    if is_initialized():
+        return
+    with _initialization_lock:
+        if is_initialized():
+            return
+        if not hasattr(torch._C, '_mps_init'):
+            raise AssertionError("Torch not compiled with MPS enabled")
+        torch._C._mps_init()
+        _initialized = True
+
+@lru_cache()
+def is_available() -> bool:
+    r"""Returns a bool indicating if MPS is currently available."""
+    if not hasattr(torch._C, '_is_mps_available'):
+        return False
+    return torch._C._is_mps_available()
```

Comment:
Why implement it again, rather than call the function from backends?
```suggestion
def is_available() -> bool:
    r"""Returns a bool indicating if MPS is currently available."""
    return torch.backends.mps.is_available()
```

Reply:
Removed is_available() and is_macos13_or_newer from torch.mps, as they already exist in torch.backends.mps.